groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovycWrapper.java

```diff
@@ -176,31 +246,63 @@ private void addAllSourcesToCompilationUnit() {
 
     private void parseAllSymbols() {
         Map<String, Set<SymbolInformation>> newFileSymbols = Maps.newHashMap();
+        Map<String, Set<SymbolInformation>> newReferences = Maps.newHashMap();
+
         unit.iterator().forEachRemaining(sourceUnit -> {
             Set<SymbolInformation> symbols = Sets.newHashSet();
             String sourcePath = sourceUnit.getSource().getURI().getPath();
+
             // This will iterate through all classes, interfaces and enums, including inner ones.
             sourceUnit.getAST().getClasses().forEach(clazz -> {
                 // Add class symbol
-                symbols.add(createSymbolInformation(clazz.getName(), getKind(clazz), createLocation(sourcePath, clazz),
-                        Optional.fromNullable(clazz.getOuterClass()).transform(ClassNode::getName)));
+                SymbolInformation classSymbol =
+                        createSymbolInformation(clazz.getName(), getKind(clazz), createLocation(sourcePath, clazz),
+                                Optional.fromNullable(clazz.getOuterClass()).transform(ClassNode::getName));
+                symbols.add(classSymbol);
+
+                // Add implemented interfaces reference
+                for (ClassNode node : clazz.getInterfaces()) {
+                    if (!node.getName().equals(GROOVY_DEFAULT_INTERFACE)) {
+                        newReferences.computeIfAbsent(node.getName(), (value) -> Sets.newHashSet()).add(classSymbol);
```

Comment:
But then I need to check to see if `putIfAbsent` return null (which is what it returns if it was absent) because if its not absent I still need to add the new reference. `putIfAbsent` and `computerIfAbsent` don't return the same thing.